tkdesigner/elements.py

```diff
@@ -0,0 +1,176 @@
+from tkinter.constants import W
+import uuid
+
+from jinja2 import Template
+from tkdesigner.template import TEMPLATE
+
+
+TEXT_INPUT_ELEMENT_TYPES = {
+    "TextArea": "Text",
+    "TextBox": "Entry"
+}
+
+
+class FigmaFrame:
+    """Represents the Figma Frame for which the GUI will be generated.
+    
+    Attributes:
+        width: Width of the window.
+        height: Height of the window.
+        asset_dir: Where the generated assets are stored.
+        elements: Child GUI elements.
+        bg_color: Color of the GUI background.
+    """
+    def __init__(self, width, height, asset_dir, bg_color="#FFFFFF", elements=[]):
+        self.width = width
+        self.height = height
+        self.asset_dir = asset_dir
+        self.elements = elements
+        self.bg_color = bg_color
+
+
+    def to_code(self, template=TEMPLATE):
+        t = Template(template)
+        return t.render(window=self, elements=self.elements)
+
+
+class FigmaElement:
+    """Represents each Figma element in the UI.
+
+    Each element has a position, size, and a unique ID.
+
+    Attributes:
+        x: X position coordinate of the element.
+        y: Y position coordinate of the element.
+        width: Width of the element.
+        height: Height of the element.
+        id_: ID unique to that element, useful for variable names.
+    """
+
+    def __init__(self, x, y, width, height):
+        self.x = x
+        self.y = y
+        self.width = width
+        self.height = height
+        self.id_ = str(uuid.uuid4())[:8]
```

Comment:
Can we use more generalised approach to naming ? 
Using a random string would make it difficult for elements to be identified. We could use the previous approach of naming the elements using their count.

Reply:
I don't mind changing that, but I think it should be a count across all elements as opposed to per element type. The reason being is that we then have to maintain a count for each element type. Let me know what you think.

Reply:
We could have a single count for all elements but it would raise confusion in the generated code. 
Eg - There would be Button1, Button and suddenly there would a Entry3. The user would think “When did I made the previous two entry 🤔”

What do you think?

Reply:
That's why I chose to do the UUIDs. We can do a count per type, not a huge deal and it would make the generated code look a bit nicer.

Reply:
Yup! Also if were to use UUID's then a better option could be Node ID. It's an unique ID given to each Figma element and is present in the API data. It's around 3-4 digit.

Reply:
What if Figma element name is used as an identifier for Tkinter objects.
The element names should be a valid Python identifier `[a-zA-Z_]\s*`  (it can be turned into one using some regex if it's invalid) and that identifier would be used to name Canvas and Rectangle objects and also the callback functions for buttons.
Callback functions can be stored in a different module that the user can then modify to make the code functional.

Reply:
The thing is - Figma allows users to have elements with same name. And the main thing on which Tkinter Designer relies on is naming. 

Which means that the name of different elements would be same. And adding checks for invalid identifiers and duplicate name would make it more complex than just a count for every single element.

Reply:
The identifiers can use a counter value or UUID as suffix. For example, `btn_submit_12` (`btn` specifies that it's a button even though the type is rectangle, `submit` is the name of button and `12` is unique identfier). The callback function can be named like `btn_submit_12_fn` or `btn_submit_12_command`.

It's more intuitive as the user doesn't have to keep in mind what elements translate to what Tkinter objects. Other elements can use similar naming like `txtbox_username_23` and `lbl_intro_2`. The prefix denotes the element type.

The package just has to be consistent with the naming scheme.

Reply:
I agree with that and I want to also add the callback function feature.

But how are we going to implement it?

Reply:
[Done](https://github.com/ParthJadhav/Tkinter-Designer/pull/67/commits/f306182a304107002ac73d0466acadec2d8e62cc)